### DISCUSSION TITLE
Plugins Error

### DISCUSSION THREAD

**Discussion by ben-demotic** [rspamd/rspamd#3760] [Q&A] [answered]
When I log into the web UI I get the following errors in the logs each time (and some nice red banners on screen).

```
Worker Type: controller
Module: csession
Message: plugin handler /plugins/selectors/list_transformss has not been found
```

```
Worker Type: controller
Module: csession
Message: plugin handler /plugins/selectors/list_extractorss has not been found
```

Could anyone point me in the right direction to resolving this?

Thanks

Ben

**Comment by moisseev** [comment, marked as answer]
Looks like you have a misconfigured reverse proxy in front of Rspamd.

- Try to connect web-browser directly to Rspamd or 
```
curl http://127.0.0.1:11334/plugins/selectors/list_transforms
curl http://127.0.0.1:11334/plugins/selectors/list_extractors
```
and see if everything is fine.

- Capture connection between reverse proxy and rspamd with tcpdump. Check if the proxy adds an extra `s` at the end of the path.

**Comment by ben-demotic** [comment]
> Looks like you have a misconfigured reverse proxy in front of Rspamd.

This was the case, but it seems some update (to Apache?) recently caused the issue.  My original reverse proxy configuration:

```
RewriteEngine on
RewriteRule ^/rspamd$ /rspamd/ [R]
ProxyPreserveHost On
ProxyPass /rspamd http://localhost:11334/
ProxyPassReverse /rspamd http://localhost:11334/
```

The corrected configuration (note the removal of the trailing forward slashes in the ProxyPass and ProxyPassReverse lines:

```
RewriteEngine on
RewriteRule ^/rspamd$ /rspamd/ [R]
ProxyPreserveHost On
ProxyPass /rspamd http://localhost:11334
ProxyPassReverse /rspamd http://localhost:11334
```

Hope this helps anyone else if they're searching for the answer to this.

Thanks to @moisseev.
